pkg/resources/nifi/secretconfig.go

```diff
@@ -699,3 +712,355 @@ func (r Reconciler) generateReadOnlyConfig(
+	var readOnlyClusterConfig map[string]string
+
+	if &r.NifiCluster.Spec.ReadOnlyConfig != (&v1.ReadOnlyConfig{}) && &r.NifiCluster.Spec.ReadOnlyConfig.BootstrapGCPProperties != (&v1.BootstrapGCPProperties{}) {
+		r.generateReadOnlyConfig(
+			&readOnlyClusterConfig,
+			r.NifiCluster.Spec.ReadOnlyConfig.BootstrapGCPProperties.OverrideSecretConfig,
+			r.NifiCluster.Spec.ReadOnlyConfig.BootstrapGCPProperties.OverrideConfigMap,
+			r.NifiCluster.Spec.ReadOnlyConfig.BootstrapGCPProperties.OverrideConfigs, log)
+	}
+
+	var readOnlyNodeConfig = map[string]string{}
+
+	for _, node := range r.NifiCluster.Spec.Nodes {
+		if node.Id == id && node.ReadOnlyConfig != nil && &node.ReadOnlyConfig.BootstrapGCPProperties != (&v1.BootstrapGCPProperties{}) {
+			r.generateReadOnlyConfig(
+				&readOnlyNodeConfig,
+				node.ReadOnlyConfig.BootstrapGCPProperties.OverrideSecretConfig,
+				node.ReadOnlyConfig.BootstrapGCPProperties.OverrideConfigMap,
+				node.ReadOnlyConfig.BootstrapGCPProperties.OverrideConfigs, log)
+			break
+		}
+	}
+
+	if err := mergo.Merge(&readOnlyNodeConfig, readOnlyClusterConfig); err != nil {
+		log.Error("error occurred during merging readonly configs",
+			zap.String("clusterName", r.NifiCluster.Name),
+			zap.Int32("nodeId", id),
+			zap.Error(err))
+	}
+
+	// Generate the Complete Configuration for the Node
+	completeConfigMap := map[string]string{}
+
+	if err := mergo.Merge(&completeConfigMap, readOnlyNodeConfig); err != nil {
+		log.Error("error occurred during merging readOnly config to complete configs",
+			zap.String("clusterName", r.NifiCluster.Name),
+			zap.Int32("nodeId", id),
+			zap.Error(err))
+	}
+
+	if len(completeConfigMap) == 0 {
+		return nil
+	}
+
+	if err := mergo.Merge(&completeConfigMap, util.ParsePropertiesFormat(r.getBootstrapGCPPropertiesConfigString(nodeConfig, id, log))); err != nil {
+		log.Error("error occurred during merging operator generated configs",
+			zap.String("clusterName", r.NifiCluster.Name),
+			zap.Int32("nodeId", id),
+			zap.Error(err))
+	}
+
+	completeConfig := []string{}
+
+	for key, value := range completeConfigMap {
+		completeConfig = append(completeConfig, fmt.Sprintf("%s=%s", key, value))
+	}
+
+	// We need to sort the config every time to avoid diffs occurred because of ranging through map
+	sort.Strings(completeConfig)
+
+	output := strings.Join(completeConfig, "\n")
+	return &output
+}
+
+func (r *Reconciler) getBootstrapGCPPropertiesConfigString(nConfig *v1.NodeConfig, id int32, log zap.Logger) string {
+	base := r.NifiCluster.Spec.ReadOnlyConfig.BootstrapGCPProperties.DeepCopy()
+	for _, node := range r.NifiCluster.Spec.Nodes {
+		if node.Id == id && node.ReadOnlyConfig != nil && &node.ReadOnlyConfig.BootstrapGCPProperties != (&v1.BootstrapGCPProperties{}) {
+			mergo.Merge(base, node.ReadOnlyConfig.BootstrapGCPProperties, mergo.WithOverride)
+		}
+	}
+
+	var out bytes.Buffer
+	t := template.Must(template.New("nConfig-config").Parse(config.BootstrapGCPPropertiesTemplate))
```

Comment:
We want to avoid maintaining templates in nifikop, right? IMO, we can return on line 761 above.

```go
if len(completeConfigMap) == 0 {
		return nil
} else {
		return completeConfigMap
}
```

Right?

Reply:
I don't know. So i should remove the template file?

Reply:
Yeah so like you did with the logback configuration here: 

https://github.com/konpyutaika/nifikop/pull/353/files#diff-5a73dc757d3f7b939a26147f59fc5df61991e53a968160047744a93828d1bbf9L363-L374

If we don't strictly **need** to maintain a template in nifikop, we shouldn't. That way we don't need to maintain it & keep it updated whenever it changes in NiFi.